testing/test_store.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+from _pytest.store import Store
+from _pytest.store import StoreKey
+
+
+def test_store() -> None:
+    store = Store()
+
+    key1 = StoreKey[str]()
+    key2 = StoreKey[int]()
+
+    # Basic functionality - single key.
+    assert key1 not in store
+    store[key1] = "hello"
+    assert key1 in store
+    assert store[key1] == "hello"
+    assert store.get(key1, None) == "hello"
+    store[key1] = "world"
+    assert store[key1] == "world"
+    # Has correct type (no mypy error).
+    store[key1] + "string"
```

Comment:
@bluetech 
I've wondered if this could be improved?
```python
    # > Cannot infer type argument 1 of "__setitem__" of "Store"
    store[key1] = 42
```

E.g. also by a runtime check, and/or better message?

Reply:
I don't think the message can be improved. Sounds like something mypy itself can improve.

Regarding runtime checks, why do you think they should be added?